### ISSUE TITLE
Swift 4?

### ISSUE THREAD

**Issue by alissonbarauna** [evermeer/EVCloudKitDao#85]
Hey guys.

Do you have any news about EVCloudKitDAO on Swift4?
I've tried to fix the errors I got, but I'm not advanced programmer and I couldn't be able to find out the solution.

These two lines contains the same error.
![image](https://user-images.githubusercontent.com/19950837/30285912-222071b2-96f6-11e7-9208-e04e08f3a590.png)

And this error is reported too.
![image](https://user-images.githubusercontent.com/19950837/30286827-5831f214-96f9-11e7-9209-cd02b03a3714.png)

I would appreciate if you guys could help me on that.

Thanks in advance.

**Comment by evermeer** [comment]
Could you please try doing a pod update?

**Comment by alissonbarauna** [comment]
The second error i mentioned has been fixed.
But the first one is still reported.

![image](https://user-images.githubusercontent.com/19950837/30291498-8c0721d0-9709-11e7-953c-9d2c034ac61a.png)

**Comment by alissonbarauna** [comment]
@evermeer The issue persists. Have you noticed it's on EVReflection?
I have commented these two lines for now!, waiting for the fixing. 

I would like to thank you for your effort for now!

**Comment by evermeer** [comment]
Strange, I have run all unit tests of EVReflection yesterday using Swift 4 (Xcode build 9 beta 6) and did not get this build error. I will have an other look at it.

**Comment by alissonbarauna** [comment]
I'm now using Xcode 9 with the last toolchain 11th September, and it works now!
I have a little question, is there a way to add more than one predicate on the queries for both local and remote database?
I haven't found  way to send predicates on an array.

**Comment by evermeer** [comment]
You can run as many EVCloudData.publicDB.connect and EVCloudData.privateDB.connect functions as you like. You could even specify a specific named CloudKit database to use for you query. If you want multiple queries for the same table (object) then you only have to change the filterId. If you look at the demo app, then you see that multiple .connect functions are started against the same Message table.

**Comment by alissonbarauna** [comment]
Really nice and good Edwin.
Just one more question, the last one i promise hehe!
Is there a way to order the results using CKLocationSortDescriptor on the .connect method, instead of just sending the name of the "field" I want to order it by.
I'm working on an app that look for places near me, and it's really important result are given from cloudkit according to my current location.

**Comment by evermeer** [comment]
Currently in the connect function you can sort on multiple fields like this:
`orderBy: Ascending(field: "Subject").Descending("creationDate")`
But I can imagine that you would like to have a custom sort. like the CKLocationSortDescriptor.
I will have a look at the OrderBy class. I think it would not be too hard to add.
I will let you know

**Comment by alissonbarauna** [comment]
That would be awesome Edwin. 
Thanks a lot for your effort on that.

**Comment by alissonbarauna** [comment]
Edwin, when I want my query to be limited in 20 records for exemple, I just need to change the value on the connect completionHandler, right?
Does it save the cursor automatically for next request?

completionHandler: { results in
            EVLog("There are \(results.count) existing news items")
            self.newsController.tableView.reloadData()
            **return results.count < 200** // Continue reading if we have less than 200 records and if there are more.
        }

**Comment by mgregory** [comment]
I have this question too. I'd like to be able to access the cursor for a query in order to stop at a specified record count and then load more records at a later time, ids the user request more of the same set (e.g. scrolls to the bottom of a table view). Currently it looks like the cursor is handled internally to `EVCloudKitDao.QueryRecords()` to continue fetching until the all the records are returned and is not exposed for use once a session is complete. Is there a way to achieve that capability at present?

**Comment by evermeer** [comment]
Currently it automatically continues fetching results as long as you return true (which is the results.count < 200). At the moment there is no mechanism to postpone this fetching by saving the cursor so that you could continue later on demand. But then why would you wait if you could just continue getting more records?

When performing a query CloudKit itself will decide how many records will be returned at maximum. Usually the amount is 100. This number can be influenced by changing `operation.resultsLimit = CKQueryOperationMaximumResults`. Currently I always use this max. The connect has no functionality to specify an other number.

**Comment by evermeer** [comment]
@alissonbarauna location sort has been added to version 3.5.0
just use the orderBy: Location(field: "yourField", relativeLocation: yourCLLocation)
